metrics/web/docs/features/public_dashboard/01_public_dashboard.md

```diff
@@ -137,3 +161,206 @@ Once we've chosen the implementation approach, let's consider the components dia
+
+The first important part is adding the ability to track the feature state to change the behavior of our application correspondingly. For this purpose, we should implement the new `PublicDashboardFeatureConfigModel` class and make changes to this module. The following subsections describes this class in a bit more detail and changes required for the feature integration.
+
+##### PublicDashboardFeatureConfigModel
+> Explain the purpose and responsibility of the class.
+
+Since we are going to store the public dashboard configuration in the Firestore database and the `FeatureConfigNotifier` manages the application state related to the feature configuration, we should be able to transfer the configuration of the public dashboard feature between different states. According to the [Presentation Layer Architecture](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/02_presentation_layer_architecture.md#model) document, to transfer the data between the states, we should use the `model` classes. So, for this purpose, we should create a `PublicDashboardFeatureConfigModel` that will be used to transfer the public dashboard feature configuration from the `FeatureConfigNotifier` to any other state in the application.
+
+##### FeatureConfig module changes
+
+The list of changes this module requires to integrate the public dashboard feature:
+- add the `isPublicDashboardFeatureEnabled` boolean field to the `FeatureConfig` model; 
+- change the `FeatureConfigData`'s `fromJson()` and `toJson()` methods, so they include handling the new added field;
+- create the [`PublicDashboardFeatureConfigModel`](#PublicDashboardFeatureConfigModel) and initialize it in the `FeatureConfigNotifier`.
+
+Let's review the following class diagram, which describes the relationship between the classes in this module:
+
+![Feature Config Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_feature_config_class_diagram.puml)
+
+#### Auth module
+
+Next, we should add the ability to sign in anonymously and detect this kind of user. For these purposes, we should create the classes and make module changes described in the following subsections.
+
+##### UserProfileViewModel
+> Explain the purpose and responsibility of the class.
+
+Since we want to disable some features on the UI if the user is anonymous, we should be able to access the user profile from the UI. As the [Presentation Layer Architecture](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/02_presentation_layer_architecture.md#view-model) document states, we should create a view model to be able to access the data from the `state` on the UI. So, we should create a view model representing a user profile to disable the `Project Groups` page and provide the ability to `Sign In` to the application for the anonymous users.
```

Comment:
🏷️ 
```suggestion
Since we want to disable some features on the UI when the user is anonymous, we should be able to access the user profile from the UI. As the [Presentation Layer Architecture](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/02_presentation_layer_architecture.md#view-model) document states, we should create a `view model` to be able to access the data from the `state` on the UI. So, we should create a view model representing a user profile to disable the `Project Groups` page and provide the ability to `Sign In` to the application for the anonymous users.
```
?